tasks.py

```diff
@@ -106,13 +113,214 @@ def update_config_file(*, thing: str, version: str):
+    response = opinionated_docker_run(
+        image=image,
+        volumes=volumes,
+        working_dir=working_dir,
+        command=command,
+        environment=environment,
+    )
+
+    if not expected_status_code(expected=1, response=response):
+        sys.exit(response["StatusCode"])
+    num_tests_ran += 1
+
+    # Ensure insecure configurations fail
+    command = "terraform plan -lock=false"
+    insecure_config_dir = TESTS_PATH.joinpath("terraform/insecure")
+    volumes = {insecure_config_dir: {"bind": working_dir, "mode": "ro"}}
+    response = opinionated_docker_run(
+        image=image,
+        volumes=volumes,
+        working_dir=working_dir,
+        command=command,
+        environment=environment,
+    )
+
+    if not expected_status_code(expected=1, response=response):
+        sys.exit(response["StatusCode"])
+    num_tests_ran += 1
+
+    # Ensure secure configurations pass
+    command = "terraform plan -lock=false"
+    secure_config_dir = TESTS_PATH.joinpath("terraform/secure")
+    volumes = {secure_config_dir: {"bind": working_dir, "mode": "ro"}}
+    response = opinionated_docker_run(
+        image=image,
+        volumes=volumes,
+        working_dir=working_dir,
+        command=command,
+        environment=environment,
+    )
+
+    if not expected_status_code(expected=0, response=response):
+        sys.exit(response["StatusCode"])
+    num_tests_ran += 1
+
+    LOG.info("%s passed %d end to end terraform tests", image, num_tests_ran)
+
+
+def run_az_stage_tests(*, image: str):
+    """Run the az tests"""
+    num_tests_ran = 0
+
+    # Ensure a basic azure help command succeeds
+    command = "az help"
+    response = opinionated_docker_run(image=image, command=command)
+
+    if not expected_status_code(expected=0, response=response):
+        sys.exit(response["StatusCode"])
+    num_tests_ran += 1
+
+    # Ensure a basic aws help command fails
+    command = "aws help"
+    response = opinionated_docker_run(image=image, command=command)
+
+    if not expected_status_code(expected=127, response=response):
+        sys.exit(response["StatusCode"])
+    num_tests_ran += 1
+
+    LOG.info("%s passed %d smoke tests", image, num_tests_ran)
+
+
+def run_aws_stage_tests(*, image: str):
+    """Run the aws tests"""
+    num_tests_ran = 0
+
+    # Ensure a basic aws help command succeeds
+    command = "aws help"
+    response = opinionated_docker_run(image=image, command=command)
+
+    if not expected_status_code(expected=0, response=response):
+        sys.exit(response["StatusCode"])
+    num_tests_ran += 1
+
+    # Ensure a basic az help command fails
+    command = "az help"
+    response = opinionated_docker_run(image=image, command=command)
+
+    if not expected_status_code(expected=127, response=response):
+        sys.exit(response["StatusCode"])
+    num_tests_ran += 1
+
+    LOG.info("%s passed %d smoke tests", image, num_tests_ran)
+
+
+def run_security_tests(*, image: str):
```

Comment:
This needs to be updated to run off of the recently built images.  Right now it pulls from docker hub, which is not accurate as of this PR.